alacritty/src/renderer/mod.rs

```diff
@@ -687,7 +697,7 @@ impl QuadRenderer {
         F: FnOnce(RenderApi<'_>) -> T,
     {
         unsafe {
-            gl::UseProgram(self.program.id);
+            gl::UseProgram(self.program.id());
```

Comment:
Curious what you'd think of a `self.program.use()` method or something like that.

I feel like if we're already creating an object for the program that provides fields and methods like `id`, we could also use that to interact with the OpenGL program in a more object-y way.

Reply:
Pair of `program.use()` and `clear_program()` doesn't really differ from pair of `gl::UseProgram`.
If we want more abstraction, they should be `self.program.with_program(|| {})`. But currently we manually control OpenGL state everywhere. I won't refactor all of these OpenGL calls entirely. Instead I want to keep the modification of this PR to be small and only related to shader creation.

BTW: `gl` itself is a low level API binding crate. If we want to make these calls more clean, we would looking for some high-level crate first. But it's out of the scope of this PR.

Reply:
> Pair of program.use() and clear_program() doesn't really differ from pair of gl::UseProgram.
If we want more abstraction, they should be self.program.with_program(|| {}).

Only because an even higher level of abstraction exists that isn't worth investing time into, doesn't mean the intermediate step isn't worth it. For consumer's it's certainly easier to see that you can just `.use()` a program, rather than having to figure out that the program's ID is what you need to pass to `gl::UseProgram`. I personally don't care *that* much, but I think your argument is pretty weak.

> we would looking for some high-level crate first

We shouldn't. There's not really a much higher level API that is suitable for Alacritty.

Reply:
All instances of `useProgram` I could find were directly after explicitly using a program with `gl::UseProgram(self.program.id())`. So it would make sense to use that `self.program` to "un-use" it again too.

This could either be done with a method on program (like `self.program.clear_program()`), or by doing both automatically using `self.program.with_program(|| { … })`.